src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,143 @@
+            switch (command) {
+            case "todo":
+                checkStringArgument(split, command);
+                String description = userInput.substring(5);
+                res = new AddCommand(new ToDo(description));
+                break;
+            case "deadline":
+                checkStringArgument(split, command);
+                String[] deadlineDetails = userInput.substring(9).split(" /by ");
+                checkDateTime(deadlineDetails);
+                String deadlineDescription = deadlineDetails[0];
+                LocalDateTime deadlineDateTime = LocalDateTime.parse(deadlineDetails[1], formatter);
+                res = new AddCommand(new Deadline(deadlineDescription, deadlineDateTime));
+                break;
+            case "event":
+                checkStringArgument(split, command);
+                String[] eventDetails = userInput.substring(6).split(" /at ");
+                checkDateTime(eventDetails);
+                String eventDescription = eventDetails[0];
+                LocalDateTime eventDateTime = LocalDateTime.parse(eventDetails[1], formatter);
+                res = new AddCommand(new Event(eventDescription, eventDateTime));
+                break;
+            case "list":
+                res = new ListCommand();
+                break;
+            case "done":
+                int completedTaskIdx = checkNumericalArgument(split, tasks);
+                res = new DoneCommand(completedTaskIdx);
+                break;
+            case "delete":
+                int taskToDelete = checkNumericalArgument(split, tasks);
+                res = new DeleteCommand(taskToDelete);
+                break;
+            case "find":
+                checkStringArgument(split, command);
+                String keyword = userInput.substring(5);
+                res = new FindCommand(keyword);
+                break;
+            case "bye":
+                res = new ExitCommand();
+                break;
+            default:
+                res = new InvalidCommand("No such command!");
+            }
+        } catch (DukeException e) {
+            res = new InvalidCommand(e.getMessage());
+        }
+        return res;
+    }
+
+    /**
+     * Checks whether the user has entered a description for the task to be added.
+     * Checks whether the user has entered a keyword for find function.
+     *
+     * @param split Array of substrings obtained from removing blank spaces from user input.
+     * @throws DukeArgumentException when user did not enter a keyword or description.
+     */
+    private static void checkStringArgument(String[] split, String command) throws DukeArgumentException {
+        if (split.length == 1 && command.equals("find")) {
+            throw new DukeArgumentException("You have not entered a keyword!");
+        } else if (split.length == 1) {
+            throw new DukeArgumentException("You have not entered a task description!");
+        } else {}

```

Comment:
If there is an empty block, is it better to comment on it to help others read your code?